tests/ipy_test_runner.py

```diff
@@ -1,205 +1,14 @@
-
-            parametrize_counter = 0
-            for kwargs in parametrized_arguments:
-                counter += 1
-                parametrize_counter += 1
-
-                # Invoke test
-                result = dict(test_module=test_module, test_method=test_method_name)
-
-                with capture() as out:
-                    try:
-                        argument_names = test_method.__code__.co_varnames[0:test_method.__code__.co_argcount]
-
-                        # Inject fixture if needed
-                        for argkey in argument_names:
-                            if argkey in kwargs:
-                                continue
-                            if argkey not in FIXTURES:
-                                raise Exception('Test method "{}" needs argument "{}" but no fixture with that name'.format(test_method_name, argkey))
-                            kwargs[argkey] = FIXTURES[argkey]()
-
-                        # Invoke test method
-                        test_method(**kwargs)
-
-                        result['result'] = '.'
-                    except:
-                        errors += 1
-                        result['result'] = 'F'
-                        result['exception'] = traceback.format_exc()
-                        result['exception_message'] = sys.exc_info()[1]
-                    finally:
-                        result['out'] = out
-                        # Unload modules loaded by the test
-                        modules_loaded_by_test = [m for m in sys.modules if m not in loaded_modules]
-                        for module_to_unload in modules_loaded_by_test:
-                            sys.modules.pop(module_to_unload)
-
-                print(result['result'], end='')
-                if result['result'] == 'F':
-                    key = '{}::{}[{}]'.format(test_module, test_method_name, parametrize_counter)
-                    collected_errors[key] = result
-
-        print()
-
-    end_time = time.time()
-
-    if errors > 0:
-        print()
-        print_title('FAILURES')
-        for key, result in collected_errors.items():
-            print_title(key, sep='_')
-            print(result['exception'])
-            needs_title = True
-            for o in result['out']:
-                if len(o):
-                    if needs_title:
-                        print_title('Captured stdout', sep='-')
-                        needs_title = False
-                    print(o)
-
-        print_title('short test summary info')
-        for key, result in collected_errors.items():
-            print('FAILED {} - {}'.format(key, result['exception_message']))
+    pytest.load_fake_module('Rhino')
+    pytest.load_fake_module('Rhino.Geometry', fake_types=['RTree', 'Sphere', 'Point3d'])
 
-    print_title('{} failed, {} passed in {:.2f}s'.format(errors, counter - errors, end_time - start_time))
-    sys.exit(errors)
+    pytest.run(HERE, ['tests/compas/files/test_base_reader.py'])
```

Comment:
i "removed" the base reader test from the test suite. renamed it to "test_base_reader.py.tmp"

Reply:
impressive simplification of the script though :)

Reply:
> fyi,
> i "removed" the base reader test from the test suite. renamed it to "test_base_reader.py.tmp"

Should dependency on pathlib/pathlib2 be removed for compatibility with IPY?

Reply:
@tetov something definitely needs to be done, because the code does not run on rhino

Reply:
@gonzalocasas will fix, tracking in #529.